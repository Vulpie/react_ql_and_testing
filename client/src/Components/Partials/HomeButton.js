import React from 'react'
import { Link } from 'react-router-dom'

import backArrowIcon from '../../assets/undo.png'

const HomeButton = () => {
	return (
		<Link to='/'>
			<img
				className='home-button'
				alt='Return button'
				src={backArrowIcon}
			/>
		</Link>
	)
}

export default HomeButton
