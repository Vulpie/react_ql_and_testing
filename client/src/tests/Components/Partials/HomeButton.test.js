import React from 'react'
import { shallow } from 'enzyme'
import HomeButton from '../../../Components/Partials/HomeButton'

// use shallow for simple components without props

test('should render HomeButton correctly', () => {
	const wrapper = shallow(<HomeButton />)
	expect(wrapper.find('.home-button').length).toBe(1) // find zwraca tablice element√≥w tak jak querySelectorAll

	expect(wrapper).toMatchSnapshot()

	// const renderer = new ReactShallowRenderer()
	// renderer.render(<HomeButton />)
	// expect(renderer.getRenderOutput()).toMatchSnapshot() // the first time it is going to pass because there is no snapshot
	// console.log(renderer.getRenderOutput())
})
