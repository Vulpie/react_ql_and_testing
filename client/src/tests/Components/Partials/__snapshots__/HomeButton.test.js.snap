// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render HomeButton correctly 1`] = `
<Link
  to="/"
>
  <p>
    Test
  </p>
  <img
    alt="Return button"
    className="home-button"
    src="undo.svg"
  />
</Link>
`;
